cinema/views.py

```diff
@@ -43,16 +66,111 @@ def get_serializer_class(self):
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset.prefetch_related()
+
+        actors = self.request.query_params.get("actors")
+        if actors:
+            actor_values = actors.split(",")
+            query = Q()
+            for value in actor_values:
+                if value.isdigit():
+                    query |= Q(id=value)
+                else:
+                    query |= Q(actors__first_name__icontains=value) | Q(
+                        actors__last_name__icontains=value
+                    )
+            queryset = queryset.filter(query)
+
+        title = self.request.query_params.get("title")
+        if title:
+            queryset = queryset.filter(
+                title__icontains=title
+            )
+
+        genres = self.request.query_params.get("genres")
+        if genres:
+            genres_list = genres.split(",")
+            print("Genres list:", genres_list)
```

Comment:
Remove the debug print statement. Debugging code should not be present in production code as it can lead to unnecessary console output.